### PR TITLE
Amb 1505 add /document reference/id endpoint

### PR DESCRIPTION
**Pull request by almas-liaqat**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: proxies/live/apiproxy/proxies/default.xml [hunk 1]
```diff
@@ -36,8 +64,14 @@
     <Condition>(request.verb = "OPTIONS") and (request.header.origin != null) and (request.header.Access-Control-Request-Method != null)</Condition>
   </RouteRule>
   <RouteRule name="NoRoutePing">
-    <Condition>(proxy.pathsuffix MatchesPath "/_ping") and ((request.verb = "GET") or (request.verb = "HEAD"))</Condition>
+    <Condition>(proxy.pathsuffix MatchesPath "/_ping") and ((request.verb = "GET") or (request.verb
```
Comment (artronics) [inline proxies/live/apiproxy/proxies/default.xml:67]: This is probably as a result of running formatter. Please make sure these are one liners.

File: sandbox/src/routes/document_reference.py [hunk 2]
```diff
@@ -1,20 +1,38 @@
 from urllib.parse import parse_qs
-
 from fastapi import APIRouter, Request
-
+from starlette.responses import Response
```
Comment (artronics) [inline sandbox/src/routes/document_reference.py:3]: We're using `fastapi`. This dependency doesn't look right. see for example https://github.com/NHSDigital/booking-and-referral-fhir-api/blob/master/sandbox/src/routes/appointment.py#L4

File: tests/test_document_reference.py [hunk 3]
```diff
@@ -31,12 +35,148 @@ def test_get_document_reference(self, get_token_client_credentials):
...
+        timestamp = int(time.time())
+        random.seed(timestamp)
+        random_integer = random.randint(0, 1000)
+        document_id += str(random_integer)
+        # Given
+        token = get_token_client_credentials["access_token"]
+        expected_post_status_code = 201
+        expected_delete_status_code = 200
+        expected_put_status_code = 404
+        request_payload = {
```
Comment (artronics) [inline tests/test_document_reference.py:97]: Just to confirm. Do we have an example of a request body in the spec? if yes then we should treat request bodies like responses i.e. loading them from a file. In this case we should load, then parse and then change the values before sending it.
If we don't have examples in the spec, please leave it for now. I think probably this is something that we need to get back to once they added more examples to the spec

File: proxies/live/apiproxy/proxies/default.xml (hunk 1 above)
Comment (almas-liaqat) [reply proxies/live/apiproxy/proxies/default.xml:67]: This is one liner

File: tests/test_document_reference.py [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
 from .configuration import config
-from .example_loader import load_example
+import base64
 
 
 class TestDocumentReference:
+    end_user_ods = "V4T0L"
+    end_user_nhs_number = "4409815415"
     target_id = "NHS0001"
+    target_identifier = json.dumps({"value": target_id, "system": "tests"})
```
[the remark this answers is not in the document]
Comment (almas-liaqat) [reply tests/test_document_reference.py:15]: Sandbox tests were failing without this header

File: tests/test_document_reference.py (hunk 3 above)
Comment (almas-liaqat) [reply tests/test_document_reference.py:91]: No as of now we dont have example

File: tests/test_document_reference.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (almas-liaqat) [reply tests/test_document_reference.py]: Removed now

**Review by ChewieCB — APPROVED** [review]

**Pull request merged**